assigner/baserepo.py

```diff
@@ -253,6 +253,28 @@ def list_commits(self, ref_name="master"):
             "/projects/{}/repository/commits".format(self.id), params
         )
 
+    def list_pushes(self):
+        return self._gl_get(
+            "/projects/{}/events?action=pushed".format(self.id)
+        )
+
+    def get_last_HEAD_commit(self, ref="master"):
+        matching_pushes = list(filter(lambda push: push['push_data']['ref'] == ref, self.list_pushes()))
+        commits = self.list_commits(ref)
+
+        HEAD = None
+
+        if commits:
+            HEAD = commits[0]
+            # Gitlab's commit created_at time uses the git metadata;
+            # rather than trusting students, we get the time the commit was pushed at
+            if matching_pushes and HEAD['id'] == matching_pushes[0]['push_data']['commit_to']:
+                # For whatever reason, Gitlab uses a different time format here than for commits...
+                unmangled_time = matching_pushes[0]['created_at'][:-1] + "-0000"
```

Comment:
Can you give an example of this? It seems like passing it through datetime may be better.

Reply:
Basically the push event time looks like `"2018-01-05T20:33:21.000Z"` whereas the commit time looks like `"2018-01-05T13:44:43.000-06:00"`. I read the `strptime` docs and I don't know how to get `datetime` to parse the first date format.